### PR TITLE
Structure definition cqf do not perform reason

### PR DESCRIPTION
**Pull request by bryantaustin13**
for [FHIR-45369](https://jira.hl7.org/browse/FHIR-45369) Do not perform Reason
StructureDefinition along with ValueSet and CodeSystem to support it.

### CODE REVIEW

File: input/definitions/Resource/CodeSystem-negation-reason-example.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "meta": {
+    "lastUpdated": "2024-07-01T07:07:38.434+11:00",
+    "profile": [
+      "http://hl7.org/fhir/StructureDefinition/shareablecodesystem"
+    ]
+  },
+  "extension": [
+    {
+      "url": "http://hl7.org/fhir/StructureDefinition/structuredefinition-wg",
+      "valueCode": "fhir"
```
Comment (brynrhodes) [inline input/definitions/Resource/CodeSystem-negation-reason-example.json:13]:
```suggestion
      "valueCode": "cqi"
```

File: input/definitions/Resource/CodeSystem-negation-reason-example.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  ],
+  "url": "http://hl7.org/fhir/CodeSystem/negation-reason-example",
+  "name": "NegationReasonExample",
+  "title": "Negation Reason Example",
+  "status": "active",
+  "experimental": false,
+  "description": "A temporary code system to avoid a qicore binding in StructureDefinition-cqf-doNotPerformReason.",
```
Comment (brynrhodes) [inline input/definitions/Resource/CodeSystem-negation-reason-example.json:21]:
```suggestion
  "description": "An example code system to illustrate the types of codes that are expected in the doNotPerformReson extension",
```

File: input/definitions/Resource/CodeSystem-negation-reason-example.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (brynrhodes) [inline input/definitions/Resource/CodeSystem-negation-reason-example.json:1]: Actually, remove this whole CodeSystem, it is not necessary, we only need to create a ValueSet example with a few of the codes from the negationReasonCodes value set in QICore

File: input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  "experimental": false,
+  "date": "2024-05-20",
+  "publisher": "HL7 International / FHIR Infrastructure",
+  "contact": [
+    {
+      "name": "Clinical Quality Information WG",
+      "telecom": [
+        {
+          "system": "url",
+          "value": "http://www.hl7.org/Special/committees/fiwg"
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json:21]:
```suggestion
          "value": "http://www.hl7.org/Special/committees/cqi"
```

File: input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    "status": "extensions"
+  },
+  "url": "http://hl7.org/fhir/StructureDefinition/cqf-doNotPerformReason",
+  "version": "5.0.0",
+  "name": "CQFDoNotPerformReason",
+  "title": "CQF Do Not Perform Reason",
+  "status": "active",
+  "experimental": false,
+  "date": "2024-05-20",
+  "publisher": "HL7 International / FHIR Infrastructure",
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json:14]:
```suggestion
  "publisher": "HL7 International / Clinical Quality Information",
```

File: input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      "uri": "http://hl7.org/v3",
+      "name": "RIM Mapping"
+    }
+  ],
+  "kind": "complex-type",
+  "abstract": false,
+  "context": [
+    {
+      "type": "element",
+      "expression": "StructureDefinition"
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json:50]: This should not be StructureDefinition, this should be limited to those resources we used it on in QICore profiles:
DeviceRequest and ServiceRequest

File: input/definitions/Resource/ValueSet-negation-reason-example.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  "url": "http://hl7.org/fhir/ValueSet/negation-reason-example",
+  "meta": {
+    "lastUpdated": "2024-07-01T07:07:38.434+11:00",
+    "profile": [
+      "http://hl7.org/fhir/StructureDefinition/shareablevalueset"
+      ]
+  },
+  "extension": [{
+    "url": "http://hl7.org/fhir/StructureDefinition/structuredefinition-wg",
+    "valueCode": "fhir"
```
Comment (brynrhodes) [inline input/definitions/Resource/ValueSet-negation-reason-example.json:13]:
```suggestion
    "valueCode": "cqi"
```

File: input/definitions/Resource/ValueSet-negation-reason-example.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  }],
+  "identifier": [
+    {
+      "system": "urn:ietf:rfc:3986",
+      "value": "urn:oid:2.16.840.1.113883.4.642.2.575"
+    }
```
Comment (brynrhodes) [inline input/definitions/Resource/ValueSet-negation-reason-example.json:19]: Was this identifier present in QICore? If so, it should be removed, this is a new extension, let the FHIR Extensions IG assign an OID for it

File: input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+      ]
+    }
+  ],
+  "description": "Indicates the reason the event was not performed.",
+  "jurisdiction": [
+    {
+      "coding": [
+        {
+          "system": "urn:iso:std:iso:3166",
+          "code": "US"
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json:32]: Remove this jurisdiction, it's not appropriate in the extensions pack

File: input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+      {
+        "id": "Extension.value[x]",
+        "path": "Extension.value[x]",
+        "type": [
+          {
+            "code": "CodeableConcept"
+          }
+        ],
+        "binding": {
+          "strength": "extensible",
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json:84]:
```suggestion
          "strength": "example",
```

File: input/definitions/Resource/ValueSet-negation-reason-example.json [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  "name": "NegationReasonExample",
+  "title": "Negation Reason Example",
+  "status": "active",
+  "experimental": false,
+  "date": "2024-07-01",
+  "description": "Codes for the reason the event did not occur or was not performed.",
+  "compose": {
+    "include": [
+      {
+        "system": "http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1021.56"
```
Comment (brynrhodes) [inline input/definitions/Resource/ValueSet-negation-reason-example.json:30]: Don't reference the VSAC value set here, include the first 3 of the specific SNOMED codes in the expansion of the QICore value set:

https://hl7.org/fhir/us/qicore/ValueSet-qicore-negation-reason.html#expansion

File: input/definitions/Resource/ValueSet-negation-reason-example.json [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+      "system": "urn:ietf:rfc:3986",
+      "value": "urn:oid:2.16.840.1.113883.4.642.2.575"
+    }
+  ],
+  "name": "NegationReasonExample",
+  "title": "Negation Reason Example",
+  "status": "active",
+  "experimental": false,
+  "date": "2024-07-01",
+  "description": "Codes for the reason the event did not occur or was not performed.",
```
Comment (brynrhodes) [inline input/definitions/Resource/ValueSet-negation-reason-example.json:26]:
```suggestion
  "description": "An example value set illustrating the kinds of codes that would be expected to be used to indicate the reason an event did not occur, or a request was not performed.",
```

**Review by brynrhodes — CHANGES_REQUESTED** [review]

**Comment by bryantaustin13** [comment]
This should be ready to be reviewed again.

File: input/definitions/Resource/ValueSet-negation-reason-example.json [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  "meta": {
+    "lastUpdated": "2024-07-01T07:07:38.434+11:00",
+    "profile": [
+      "http://hl7.org/fhir/StructureDefinition/shareablevalueset"
+    ]
+  },
+  "extension": [
+    {
+      "url": "http://hl7.org/fhir/StructureDefinition/structuredefinition-wg",
+      "valueCode": "fhir"
```
Comment (brynrhodes) [inline input/definitions/Resource/ValueSet-negation-reason-example.json:14]:
```suggestion
      "valueCode": "cqi"
```

File: input/definitions/Resource/ValueSet-negation-reason-example.json [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    }
+  ],
+  "name": "NegationReasonExample",
+  "title": "Negation Reason Example",
+  "status": "active",
+  "experimental": false,
+  "date": "2024-07-01",
+  "description": "An example value set illustrating the kinds of codes that would be expected to be used to indicate the reason an event did not occur, or a request was not performed.",
```
Comment (brynrhodes) [inline input/definitions/Resource/ValueSet-negation-reason-example.json:22]:
```suggestion
  "description": "An example value set illustrating the kinds of codes that would be expected to be used to indicate the reason an event did not occur, or a request was not performed.",
  "publisher" : "HL7 International / Clinical Quality Information",
  "contact": [
    {
      "name" : "Clinical Quality Information WG",
      "telecom": [
        {
          "system": "url",
          "value" : "http://www.hl7.org/Special/committees/cqi"
        }
      ]
    }
  ],
```

File: input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+      "uri": "http://hl7.org/v3",
+      "name": "RIM Mapping"
+    }
+  ],
+  "kind": "complex-type",
+  "abstract": false,
+  "context": [
+    {
+      "type": "element",
+      "expression": "StructureDefinition"
```
Comment (brynrhodes) [inline input/definitions/Resource/StructureDefinition-cqf-doNotPerformReason.json:44]:
```suggestion
      "expression": "DeviceRequest"
    }, {
      "type": "element",
      "expression": "ServiceRequest"
```

**Review by brynrhodes — CHANGES_REQUESTED** [review]

**Review by brynrhodes — APPROVED** [review]

**Comment by brynrhodes** [comment]
Based on discussion with FHIR-I, this extension is not required, the `reason` element of the ServiceRequest and DeviceRequest resources should be used to capture the reason for not performing the activity.

**Pull request closed without merging**